Ex3/Ex3.java

```diff
@@ -0,0 +1,128 @@
+import java.util.TreeSet;
+
+public class Ex3 {
+	
+	private int rows;
+	private int columns;
+	private char[][] matrix;
+	private Dictionary dictionary;
+	
+	//a set for final valid words in the dictionary that were found in the grid
+	private TreeSet<String> wordsInMatrix = new TreeSet<>();
+	
+	/**
+	 * Constructor
+	 * @param rows number of rows
+	 * @param columns number of columns
+	 * @param matrix Matrix of rows x columns filled with characters of type char
+	 * @param dictionary a Dictionary object
+	 */
+	public Ex3(int rows, int columns, char[][] matrix, Dictionary dictionary) {
+		this.rows = rows;
+		this.columns = columns;
+		this.matrix = matrix;
+		this.dictionary = dictionary;
+
+		wordsInMatrix = findWords(rows, columns, matrix, dictionary);
+	}
+
+	/**
+	 * A method that iterates over the characters in a matrix and
+	 * initiate on each cell the check-up
+	 * @param rows
+	 * @param columns
+	 * @param matrix
+	 * @param dictionary
+	 * @return a set of all words in found in the grid
+	 */
+	private TreeSet<String> findWords(int rows, int columns, char[][] matrix, Dictionary dictionary) {
+		for (int i = 0; i < matrix.length; i++) {
+			for (int j = 0; j < matrix[i].length; j++) {
+				checkPath(i,j, new StringBuilder(), new boolean[rows][columns]);
+			}
+		}	
+		return this.wordsInMatrix;
+	}
+	
+	/**
+	 * Check if a valid path going out of this cell
```

Comment:
It's always good to say what a method (especially one with a name like checkXXX) does with the result. This has no return type, and no declared exceptions, so someone has to read the implementation to see what is going on, they can't just read the javadoc.